### PR TITLE
[inference] add smooth function and delete useless code for smoothquant

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- Add smooth function for smoothquant
- Delete useless code

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/quant/smoothquant/models/base_model.py [hunk 1]
```diff
@@ -22,53 +26,10 @@
 from ....tensor_parallel.kvcache_manager import MemoryManager
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/base_model.py:26]: Though not modifications made in this PR, maybe it's better to import from colossalai.inferece.XXX

File: colossalai/inference/quant/smoothquant/models/base_model.py [hunk 2]
```diff
@@ -166,6 +127,84 @@ def prepare_inputs_for_generation(self, *args, **kwargs):
...
 
+    def collect_act_scales(self, model, tokenizer, dataset_path, device, num_samples=512, seq_len=512):
+        dataset = load_dataset("json", data_files=dataset_path, split="train")
+        dataset = dataset.shuffle(seed=42)
+
+        for i in tqdm(range(num_samples)):
+            input_ids = tokenizer(
+                dataset["rows"][0][i]["row"]["text"], return_tensors="pt", max_length=seq_len, truncation=True
+            ).input_ids.to(device)
+            model(input_ids)
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/base_model.py:138]: Wonder what's the purpose of `model(input_ids)` here. Is it going to be assigned to some variable?

File: colossalai/inference/quant/smoothquant/models/base_model.py [hunk 3]
```diff
@@ -166,6 +127,84 @@ def prepare_inputs_for_generation(self, *args, **kwargs):
...
+        dataset = load_dataset("json", data_files=dataset_path, split="train")
+        dataset = dataset.shuffle(seed=42)
+        for i in pbar:
+            input_ids = tokenizer(
+                dataset["rows"][0][i]["row"]["text"],
+                return_tensors="pt",
+                max_length=seq_len,
+                truncation=True,
+            ).input_ids.to(device)
+            model(input_ids)
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/base_model.py:151]: Same problem as above

File: colossalai/inference/quant/smoothquant/models/llama.py [hunk 4]
```diff
@@ -756,15 +753,15 @@ class SmoothLlamaForCausalLM(BaseSmoothForCausalLM):
...
+    def get_act_dict(
         self,
         tokenizer,
         dataset_path,
         num_samples=512,
         seq_len=512,
     ):
         llama_model = self.model
-        llama_config = llama_model.config
+        llama_model.config
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/llama.py:764]: What's this used for?

File: colossalai/inference/quant/smoothquant/models/llama.py [hunk 5]
```diff
@@ -7,14 +7,11 @@
 from functools import partial
 from typing import List, Optional, Tuple, Union
 
-import numpy as np
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
-from datasets import load_dataset
 from torch import nn
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/llama.py:13]: duplicated import

File: colossalai/inference/quant/smoothquant/models/base_model.py (hunk 2 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/base_model.py:138]: Get information in  statistic function registerd in model.

File: colossalai/inference/quant/smoothquant/models/base_model.py (hunk 3 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/base_model.py:151]: See above.

File: colossalai/inference/quant/smoothquant/models/llama.py (hunk 5 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/llama.py:13]: Removed.

File: colossalai/inference/quant/smoothquant/models/llama.py (hunk 4 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/llama.py:764]: Removed.

File: colossalai/inference/quant/smoothquant/models/base_model.py (hunk 1 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/base_model.py]: Updated.

**Review by yuanheng-zhao — APPROVED** [review]

**Pull request merged**